staging/src/k8s.io/dynamic-resource-allocation/structured/allocator_test.go

```diff
@@ -716,6 +718,42 @@ func TestAllocator(t *testing.T) {
 
 			expectResults: nil,
 		},
+		"admin-access": {
+			claimsToAllocate: func() []*resourceapi.ResourceClaim {
+				c := claim(claim0, req0, classA)
+				c.Spec.Devices.Requests[0].AdminAccess = true
+				return []*resourceapi.ResourceClaim{c}
+			}(),
+			allocatedClaims: objects(
+				allocatedClaim(claim0, req0, classA,
+					deviceAllocationResult(req0, driverA, pool1, device1, false),
+					deviceAllocationResult(req1, driverA, pool1, device2, false),
+				),
+			),
+			classes: objects(class(classA, driverA)),
+			slices:  objects(sliceWithOneDevice(slice1, node1, pool1, driverA)),
+			node:    node(node1, region1),
+
+			expectResults: []any{
+				allocationResult(localNodeSelector(node1), deviceAllocationResult(req0, driverA, pool1, device1, true)),
+			},
+		},
+		"already-allocated-for-admin-access": {
+			claimsToAllocate: objects(claim(claim0, req0, classA)),
+			allocatedClaims: objects(
+				allocatedClaim(claim0, req0, classA,
+					deviceAllocationResult(req0, driverA, pool1, device1, true),
+					deviceAllocationResult(req1, driverA, pool1, device2, true),
+				),
+			),
+			classes: objects(class(classA, driverA)),
+			slices:  objects(sliceWithOneDevice(slice1, node1, pool1, driverA)),
+			node:    node(node1, region1),
+
+			expectResults: []any{
+				allocationResult(localNodeSelector(node1), deviceAllocationResult(req0, driverA, pool1, device1, false)),
```

Comment:
This seems counter intuitive to me. Why would you not want to report that this the allocation has admin access?

Reply:
The *new* request in `claimsToAllocate` isn't using admin access, so the expected result doesn't either.

The *old* allocations in `allocatedClaims` use it, therefore they don't block using the device for a normal claim.